src/common/src/array/stream_chunk_iter.rs

```diff
@@ -12,137 +12,97 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use super::column::Column;
-use super::Row;
+use std::iter::once;
+
+use auto_enums::auto_enum;
+
+use super::RowRef;
 use crate::array::{Op, StreamChunk};
-use crate::types::{DatumRef, ToOwnedDatum};
 
 impl StreamChunk {
-    pub fn rows(&self) -> impl Iterator<Item = RowRef<'_>> {
+    /// Return an iterator on stream records of this stream chunk.
+    pub fn records(&self) -> impl Iterator<Item = RecordRef<'_>> {
         StreamChunkRefIter {
             chunk: self,
-            idx: 0,
+            inner: self.data_chunk.rows(),
         }
     }
+
+    /// Return an iterator on rows of this stream chunk.
+    ///
+    /// Should consider using [`StreamChunk::records`] if possible.
+    pub fn rows(&self) -> impl Iterator<Item = (Op, RowRef<'_>)> {
+        self.records().flat_map(|r| r.into_row_refs())
+    }
```

Comment:
There are additional cost when `.record()` and `flat_map` it back. Can we simply iter the data chunk?

Reply:
Fixed.